pkg/storaged/details.js

```diff
@@ -129,6 +166,28 @@
                     filesystem_options.push(entry);
             }
 
+            /*
+             * Older Udisks/storaged implementations don't have CreateAndFormatPartition
+             * so we simulate it. Note that in addition, the older Udisks does not fill defaults
+             * for various configuration automatically. So we make the same or similar guesses
+             * and fill them in here.
+             */
+            function create_and_format_partition(block_ptable, start, size, type, options, config_items) {
+                return block_ptable.CreatePartition(start, size, "", "", { })
+                    .then(function(partition) {
+                        var block = client.blocks[partition];
+                        var promises = config_items.map(function(item) {
+                            item = complete_configuration_item(block, item);
+                            return block.AddConfigurationItem(item, { });
+                        });
+                        return cockpit.all(promises).then(function() {
+                            return block.Format(type, options).then(function() {
```

Comment:
Format will change the `block.IdUUID` property and we want to put the new value into fstab and crypttab, not the old one. Also Format might create a new cleartext block device and we need to fall back to that if there is no UUID.  (Hmm, I wonder if udisks/storaged has the same bug, actually.)

So AddConfigurationItem needs to happen after the Format.